extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/TransitKeyConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class TransitKeyConfig {
+
+    /**
+     * Specifies the name of the key to use. By default this will be the property key alias. Used when
+     * the same transit key is used with different configurations. Such as:
+     * 
+     * <pre>
+     *     {@code
+     *     quarkus.vault.transit.key.foo1.name=foo
+     *     quarkus.vault.transit.key.foo1.key_version=1
+     *     quarkus.vault.transit.key.foo2.name=foo
+     *     quarkus.vault.transit.key.foo2.key_version=2
+     *     }
+     * </pre>
+     */
+    @ConfigItem
+    public Optional<String> name;
+
+    // sign
+
+    /**
+     * Set to true when the input is already hashed.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#prehashed">api documentation for prehashed</a>
+     */
+    @ConfigItem
+    public Optional<Boolean> prehashed;
```

Comment:
I think we should drop it until we support a `hash` directly at the main interface level, but even in that case I'm not sure we'd want to do 2 calls (first prehash and then sign) instead of one.

Reply:
I think it is more the case where your quarkus app is a signing service for inputs that you know have already been hashed by somebody else. instead of passing the entire (large) document to sign, you only pass the hash to quarkus (through a messaging service for instnce), which is far more efficient. 
I thought this would be a value added of the extension to allow setting that up into the configuration for a specific key.
let me know if you feel strong about not supporting this today.

Reply:
@vsevel OK, I guess may be it will work if the hash has been prepared outside of Vault, you are right, lets keep it